flags/mod.ts

```diff
@@ -3,17 +3,96 @@
 
 import { assert } from "../_util/assert.ts";
 
+type Id<T> = T extends Record<string, unknown>
+  ? T extends infer U ? { [K in keyof U]: Id<U[K]> } : never
+  : T;
+
+type UnionToIntersection<T> =
+  (T extends unknown ? (args: T) => unknown : never) extends
+    (args: infer R) => unknown ? R extends Record<string, unknown> ? R : never
+    : never;
+
+type BooleanType = boolean | string | undefined;
+type StringType = string | undefined;
+type ArgType = BooleanType | StringType;
+
+type Values<
+  B extends BooleanType,
+  S extends StringType,
+  D extends Record<string, unknown> | undefined,
+> = undefined extends ((false extends B ? undefined : B) & S) ? // deno-lint-ignore no-explicit-any
+Record<string, any>
```

Comment:
@kt3k Currently I use `Record<string, any>` as the default type for the arguments when no types are defined in the options object. Should we change this to `Record<string, unknown>` or do we watn to keep `Record<string, any>`?

Reply:
yes, would be `string | number | boolean | undefined`.
But I also think that `any` should be sufficient in this case. If someone wants it type-safe, then he would specify option types anyway.

Reply:
I agree, `any` seems like a safe bet here